File: backEnd/BookStore.Domain/Entities/Order.cs
using BookStore.Domain.Constants;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace BookStore.Domain.Entities;

public class Order
{
    public int OrderId { get; set; }
    public DateTime OrderDate { get; set; }
    public decimal TotalAmount { get; set; }
    public OrderStatus Status { get; set; }
    public string? StripeSessionId { get; set; }
    public int? SavedAddressId { get; set; }
    public Address? SavedAddress { get; set; }
    public ShippingAddressSnapshot ShippingAddress { get; set; } = new();

    public int CustomerId { get; set; }
    public Customer? Customer { get; set; }

    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    public Payment? Payment { get; set; }
    public Shipping? Shipping { get; set; }
}

File: frontEnd/src/app/models/Orders/order.ts
import { Payment } from "../Payments/payment";
import { OrderItem } from "./order-item";
import { OrderStatus } from "./order-status";

export interface ShippingAddressDto {
  recipientName: string;
  recipientPhone: string;
  addressLine1: string;
  addressLine2?: string | null;
  city: string;
  state?: string | null;
  postalCode: string;
  country: string;
}
export interface Order {
  orderId: number;
  orderDate: string; // ISO
  totalAmount: number;
  status: OrderStatus;
  stripeSessionId?: string | null;

  shippingAddress: ShippingAddressDto;

  customerId: number;
  orderItems: OrderItem[];
  payment?: Payment | null;
}
